specs/interop/superchain-weth.md

```diff
@@ -0,0 +1,107 @@
+# Superchain WETH
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+Superchain WETH is a version of the standard WETH contract that allows ETH to be interoperable across the Superchain.
+Superchain WETH treats ETH as an ERC-20 token for interoperability and avoids native ETH support.
+Superchain WETH also introduces a liquidity contract used to provide liquidity for native ETH across chains.
+
+## SuperchainWETH
+
+```solidity
+contract SuperchainWETH is WETH98, ISuperchainERC20 {
+  L1Block internal l1Block = L1Block(Predeploys.L1_BLOCK);
+  L2ToL2CrossDomainMessenger messenger = L2ToL2CrossDomainMessenger(Predeploys.L2_TO_L2_CROSSDOMAIN_MESSENGER);
+  ETHLiquidity liquidity = ETHLiquidity(Predeploys.ETH_LIQUIDITY);
+
+  function deposit() public payable override {
+    if (l1Block.isCustomGasToken()) revert IsCustomGasToken();
+    super.deposit();
+  }
+
+  function withdraw(uint256 wad) public override {
+    if (l1Block.isCustomGasToken()) revert IsCustomGasToken();
+    super.withdraw(wad);
+  }
+
+  function bridgeERC20(uint256 wad, uint256 chainId) external {
+    _bridgeERC20To(msg.sender, wad, chainId, true);
+  }
+
+  function bridgeERC20To(address to, uint256 wad, uint256 chainId) external {
+    _bridgeERC20To(to, wad, chainId, true);
+  }
+
+  function bridgeERC20Wrapped(uint256 wad, uint256 chainId) external {
```

Comment:
We will want to be mindful of the ABI as we will want to implement superchain erc20, which the latest is here: https://github.com/ethereum-optimism/specs/pull/71#discussion_r1631562710

There may be additional functions that would be nice to have but we should be careful about expanding the ABI too much as it increases loc and therefore chances of bugs